### PR TITLE
feat(pageserver): warn on aux v1 tenants + default to v2

### PR DESCRIPTION
**Pull request by skyzh**
## Problem

part of https://github.com/neondatabase/neon/issues/8623

We want to discover potential aux v1 customers that we might have missed from the migrations.

## Summary of changes

Log warnings on basebackup, load timeline, and the first put_file.

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: pageserver/src/pgdatadir_mapping.rs [hunk 1]
```diff
@@ -755,7 +755,13 @@ impl Timeline {
     ) -> Result<HashMap<String, Bytes>, PageReconstructError> {
         let current_policy = self.last_aux_file_policy.load();
         match current_policy {
-            Some(AuxFilePolicy::V1) | None => self.list_aux_files_v1(lsn, ctx).await,
+            Some(AuxFilePolicy::V1) | None => {
+                warn!(
+                    "tenant={} timeline={} is using deprecated aux file policy V1",
```
Comment (arpad-m) [inline pageserver/src/pgdatadir_mapping.rs:760]: shouldn't tenant and timeline be already part of the span?

File: pageserver/src/pgdatadir_mapping.rs [hunk 2]
```diff
@@ -1605,6 +1611,10 @@ impl<'a> DatadirModification<'a> {
                 if aux_files_key_v1.is_empty() {
                     None
                 } else {
+                    warn!(
+                        "tenant={} timeline={} is using deprecated aux file policy V1",
```
Comment (arpad-m) [inline pageserver/src/pgdatadir_mapping.rs:1615]: same here

**Review by jcsp — APPROVED** [review]

**Comment by arpad-m** [comment]
test failure is legit:
```
--- TRY 4 STDERR:        pageserver tenant::tests::test_branch_copies_dirty_aux_file_flag ---
thread 'tenant::tests::test_branch_copies_dirty_aux_file_flag' panicked at pageserver/src/tenant.rs:6000:9:
assertion `left == right` failed: aux file is written with switch_aux_file_policy unset (which is v1), so we should keep v1
```

**Comment by skyzh** [comment]
yep, this doesn't pass unit tests b/c the unit tests assume the default is v1. I'll need to refactor the unit test or drop it in order to make it work.

**Comment by skyzh** [comment]
fixed unit tests, will merge once CI passes

File: pageserver/src/pgdatadir_mapping.rs (hunk 2 above)
Comment (skyzh) [reply pageserver/src/pgdatadir_mapping.rs]: https://github.com/neondatabase/neon/pull/8625/commits/cb91592132afa9b31310cce52f99425351726167

File: pageserver/src/pgdatadir_mapping.rs (hunk 1 above)
Comment (skyzh) [reply pageserver/src/pgdatadir_mapping.rs]: https://github.com/neondatabase/neon/pull/8625/commits/cb91592132afa9b31310cce52f99425351726167

**Review by arpad-m — APPROVED** [review]

**Pull request merged**